src/openzaak/import_data/views.py

```diff
@@ -217,3 +217,25 @@ def retrieve(self, request, *args, **kwargs):
             attachment=True,
             mimetype="text/csv",
         )
+
+
+class ImportDestroyView(mixins.DestroyModelMixin, viewsets.GenericViewSet):
+    import_type: ImportTypeChoices
+    permission_classes = (ImportAuthRequired,)
+    lookup_field = "uuid"
+
+    def get_queryset(self):
+        return Import.objects.filter(import_type=self.import_type)
+
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+
+        if instance.status not in ImportStatusChoices.deletion_choices:
+            message = (
+                _("Import cannot be deleted due to current status: %s")
+                % instance.status
```

Comment:
It's not really an issue in this case, but in general I think it is better to use `.format` with keyword arguments for translatable strings. That way, if there are multiple variables used is it possible to know which is which and you can move them around in the translated version if needed

```suggestion
                _("Import cannot be deleted due to current status: {status}").format(status=instance.status)
```

Reply:
I usually use positional string interpolation for string formatting with translations for only one variable. [This should always be compatible with translations ](interpolation)(be it lazy or not).